docs/UpgradeGuide.md

```diff
@@ -0,0 +1,568 @@
+# Upgrading ddtrace
+
+# From 0.x to 1.0
```

Comment:
Minor: Should we just merge these headers?

Reply:
There's a link to this header from the `GettingStarted.md`. Would be better to keep this header for the future, when we have more upgrades to list.